### ISSUE TITLE
CONAN_LOGIN_USERNAME and CONAN_PASSWORD are ignored in non-interactive mode

### ISSUE THREAD

**Issue by ytimenkov** [conan-io/conan#2867]
Conan 1.3.2

To reproduce:
1. Run `conan user -c all`
2. Set `CONAN_LOGIN_USERNAME`, `CONAN_PASSWORD` and `CONAN_NON_INTERACTIVE` environment variables.
3. Run `conan upload -r staging` to remove which requires authentication.

**Expected result**: Conan uses credentials stored in environment variables (as was in Conan 1.2)
**Actual result**: Upload fails with message:
```
Please log in to "staging" to perform this action. Execute "conan user" command.
ERROR: Conan interactive mode disabled. [Remote: staging]
```

My best guess is that in `userio.py`
```python
    def request_login(self, remote_name, username=None):
        """Request user to input their name and password
        :param username If username is specified it only request password"""
        self._raise_if_non_interactive()
```
call to `self._raise_if_non_interactive()` is unnecessary as this check performed by functions called from it (Having prompt printed is not that ugly, or move this check closer to `conan user` command).

Option to always require explicit `conan user` is also possible, although feels little bit inconsistent and makes use of credentials variables pointless.

**Comment by lasote** [comment]
You are right, thanks for the perfect analysis of the issue, that is exactly the problem.
